### PR TITLE
Add accessibility improvements in blog post section

### PR DESCRIPTION
**Pull request by tauthomas01**
While working on Debut's accessibility issues, I want to apply the same improvements on Starter Theme. This PR focuses on blog post :

- Removed the second <a> link in the article so we can avoid duplicating links
- Added `aria label` for "Read more"
- Used `<ul>` and `<li>` instead of `<div>` so the user can navigate easier with a hierarchy type of list.
- Added `<ul>` and `<li>` for the list of tags
- Replaced `<h2>` to `<h3>` in the article's title

[Demo store](https://064qpumzdmtzpd80-662601789.shopifypreview.com/)
[Theme editor](https://tauclothes.myshopify.com/admin/themes/15250030653/editor)

References to Debut's accessibility PR improvements
https://github.com/Shopify/debut/pull/390
https://github.com/Shopify/debut/pull/393

### CODE REVIEW

**Comment by tauthomas01** [comment]
@t-kelly Thanks! I added some comments

File: src/sections/blog-posts.liquid [hunk 1]
```diff
@@ -11,74 +11,90 @@
...
-            {{ article.excerpt }}
-          {% else %}
-            {{ article.content | strip_html | truncate: 150 }}
+          {% if article.tags.size > 0 %}
+            {%- comment -%}
+              The Tag link list functions as a list of content items.
+              Non-visual users will expect lists of content to appear
+              in a list structure, allowing for easier understanding and navigation of list items.
+            {%- endcomment -%}
+            <ul>
```
Comment (huguestennier) [inline src/sections/blog-posts.liquid]: Can you add an `aria-label="tags"` here so when we navigate to the list we get some context ?

Comment (tauthomas01) [reply src/sections/blog-posts.liquid]: Good point, thanks!

File: src/sections/blog-posts.liquid [hunk 2]
```diff
@@ -11,74 +11,90 @@
...
+    <h2>{{ section.settings.title | escape }}</h2>
   {% endif %}
 
   {%- assign blog = blogs[section.settings.blog] -%}
 
-  {% for article in blog.articles limit: section.settings.post_limit %}
-    <div class="blog-post__item">
-      {% if article.image %}
+  {%- comment -%}
+    Due to accessibility requirement, the Blog entries list functions
```
Comment (chrisberthe) [inline src/sections/blog-posts.liquid]: "requirement**s**"

File: src/sections/blog-posts.liquid [hunk 3]
```diff
@@ -11,74 +11,90 @@
...
+    as a list of content items. Non-visual users will expect lists of content
+    to appear in a list structure, allowing for easier understanding and navigation of list items.
+  {%- endcomment -%}
+  <ul class="blog-post">
+    {% for article in blog.articles limit: section.settings.post_limit %}
+      <li class="blog-post__item">
+        {%- comment -%}
+          Due to accessibility requirement, we are regrouping all logical elements into the <a> element.
```
Comment (chrisberthe) [inline src/sections/blog-posts.liquid]: Same here.

**Comment by tauthomas01** [comment]
If my CircleCi reports an error, how can I fix it?

**Comment by chrisberthe** [comment]
We're still working on updating our CI system, i.e. removing CircleCI - so you can merge it even if it fails.

**Comment by chrisberthe** [comment]
If you want to see if there's any linting issues on your side, run `yarn lint` in your terminal before merging.

**Comment by tauthomas01** [comment]
Thanks for the reminder @chrisberthe , looks good on my end.

**Pull request merged**